jsonrpc/src/common/messageReader.ts

```diff
@@ -202,13 +215,9 @@ export class ReadableStreamMessageReader extends AbstractMessageReader {
 		this.nextMessageLength = -1;
 		this.messageToken = 0;
 		this.partialMessageTimer = undefined;
-		this.callback = callback;
-		const result = this.readable.onData((data: Uint8Array) => {
-			this.onData(data);
-		});
 		this.readable.onError((error: any) => this.fireError(error));
```

Comment:
What do we do with these. The leaked from the beginning but my internal model was always that you only use a reader once. Since this is a wrapper around a readable stream would an easier fix not simply be to create a new reader and dispose the old one instead of making this a full event emitter.

Can you explain why you want to have two listen calls?

Reply:
See my comment here for alternative (throw exception option) https://github.com/microsoft/vscode-languageserver-node/issues/1568#issuecomment-2408004113